relay-monitors/src/lib.rs

```diff
@@ -176,7 +193,22 @@ pub fn process_check_in(payload: &[u8]) -> Result<Vec<u8>, ProcessCheckInError>
         return Err(ProcessCheckInError::InvalidEnvironment);
     }
 
-    Ok(serde_json::to_vec(&check_in)?)
+    static NAMESPACE: OnceCell<Uuid> = OnceCell::new();
+    let namespace = NAMESPACE
+        .get_or_init(|| Uuid::new_v5(&Uuid::NAMESPACE_URL, b"https://sentry.io/crons/#did"));
+
+    // Use the project_id + monitor_slug as the routing key hint. This helps ensure monitor
+    // check-ins are processed in order by consistently routing check-ins from the same monitor.
+
+    let slug = &check_in.monitor_slug;
+    let project_id_slug_key = format!("{project_id}:{slug}");
+
+    let routing_hint = Uuid::new_v5(namespace, project_id_slug_key.as_bytes());
```

Comment:
The most efficient would be to instantiate a custom SHA1 hasher here and hash the three values in sequence without allocating. The `uuid` crate provides a `Builder` helper but it does not provide this functionality unfortunately. TL;DR I think allocating here is fine. I simplified a bit to and added a separator, such that for example `(1, '1pass')` and `(11, 'pass')` map to different UUIDs.

Reply:
cool. Yeah this is actually what I originally had haha